pkg/rpcclient/wsclient.go

```diff
@@ -394,102 +585,172 @@ func (c *WSClient) performUnsubscription(id string) error {
 
 // SubscribeForNewBlocks adds subscription for new block events to this instance
 // of the client. It can be filtered by primary consensus node index and/or block
-// index allowing to receive blocks since the specified index only, nil value is
-// treated as missing filter.
+// index allowing to receive blocks since/till the specified index only, nil value
+// is treated as missing filter.
 //
-// Deprecated: please, use SubscribeForNewBlocksWithChan. This method will be removed in future versions.
+// Deprecated: please, use ReceiveBlocks. This method will be removed in future versions.
 func (c *WSClient) SubscribeForNewBlocks(primary *int, sinceIndex, tillIndex *uint32) (string, error) {
-	return c.SubscribeForNewBlocksWithChan(primary, sinceIndex, tillIndex, c.Notifications)
+	var flt *neorpc.BlockFilter
+	if primary != nil || sinceIndex != nil || tillIndex != nil {
+		flt = &neorpc.BlockFilter{Primary: primary, Since: sinceIndex, Till: tillIndex}
+	}
+	ch := make(chan *block.Block)
+	go func() {
+	waitloop:
+		for {
+			select {
+			case b, ok := <-ch:
+				if !ok {
+					c.Notifications <- Notification{
+						Type: neorpc.MissedEventID,
+					}
+					return
+				}
+				c.Notifications <- Notification{
+					Type:  neorpc.BlockEventID,
+					Value: b,
+				}
+			case <-c.done:
+				break waitloop
+			}
+		}
+		for range ch {
+		}
+	}()
+	return c.ReceiveBlocks(flt, ch)
```

Comment:
Why should we bother with type-specific channels internally for old APIs? The old behavior is to push _everything_ into `c.Notifications` and it's easy to keep it alongside new logic.

Reply:
Fixed.